src/components/MessageInput/index.js

```diff
@@ -0,0 +1,144 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import styles from './styles.scss';
+import i18n from './i18n';
+
+const UIHeightOffset = 23;
+// the extra height of the entire field with paddings and borders
+
+export default class MessageInput extends Component {
+  static propTypes = {
+    value: PropTypes.string.isRequired,
+    currentLocale: PropTypes.string.isRequired,
+    disabled: PropTypes.bool,
+    minHeight: PropTypes.number,
+    maxHeight: PropTypes.number,
+    maxLength: PropTypes.number,
+    onSend: PropTypes.func,
+    onChange: PropTypes.func,
+    onHeightChange: PropTypes.func,
+  }
+  static defaultProps = {
+    disabled: false,
+    onSend: undefined,
+    onChange: undefined,
+    onHeightChange: undefined,
+    minHeight: 63,
+    maxHeight: 300,
+    maxLength: 5000,
+  }
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      value: props.value,
+      height: props.minHeight,
+    };
```

Comment:
do we miss the newHeight calculation here when component is mounted at first time with value passed in?

Reply:
Add newHeight calculation in componentDidMount in case component is mounted with existing value.